model/project_parser.go

```diff
@@ -1024,6 +1025,17 @@ func TranslateProject(pp *ParserProject) (*Project, error) {
 	proj.Tasks, proj.TaskGroups, errs = evaluateTaskUnits(tse, tgse, vse, pp.Tasks, pp.TaskGroups)
 	catcher.Extend(errs)
 
+	for _, bv := range pp.BuildVariants {
+		for _, t := range bv.Tasks {
+			// Ignore errors here because we're only checking if the selector is empty- errors are handled in evaluateBuildVariants.
```

Comment:
I moved it over to build variants so it can be layered in with the other processing! Now it doesn't reloop over the buildvariants & tasks, ty for the suggestion!

We can't do this logic in the validator since the raw task selectors don't get translated over to the BuildVariant struct, only the finalized tasks (so if a selector selected none, we wouldn't know that it even existed or what to tell the user)